### DISCUSSION TITLE
if statement + uses

### DISCUSSION THREAD

**Discussion by andrewbihl** [community/community#45370] [Actions]
### Select Topic Area

Bug

### Body

It appears that `if` conditionals are skipped when using a reusable workflow. I've attached a screenshot which demonstrates this.

Here is a link to the Github documentation which clearly outlines that this should work: https://docs.github.com/en/actions/learn-github-actions/expressions#example-expression-in-an-if-conditional

<img width="435" alt="Screen Shot 2023-01-25 at 11 52 01 AM" src="https://user-images.githubusercontent.com/16709744/214674545-3950a844-de9a-47fa-9dfc-cdef53881fda.png">

As always, Github Actions continues to disappoint in terms of its syntax, documentation, support, and overall UX.

**Comment by airtower-luna** [comment]
Only the part in `${{ ... }}` gets evaluated as an expression, everything else is a string literal. You wrapped only the variable in `${{ ... }}`, which means the resulting value is always a string ending in ` == 'false'`. And when the Actions runner converts string to boolean *any* non-empty string becomes `true`. It's a common mistake. :sweat_smile: 

Either of the following should work:

```yaml
if: ${{ vars.LOCK_PROD_DEPLOYS == 'false' }}
```

```yaml
if: vars.LOCK_PROD_DEPLOYS == 'false'
```

Without any `${{ ... }}` (the latter option) the whole string is implicitly treated as an expression.
